### PR TITLE
🐛 Fix connectivity check order of the args, pin core version to 1.4.6

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
kind/bug

**What this PR does / why we need it**:
We need to use 1.4.4 version of CAPI for the initial release to avoid potential issues with rancher integration.
Fixes wrong argument order in cluster connectivity check.

Also adds a guide for default go.work file to make e2e tests resolvable in the current environment.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Addresses #120 

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits into logical changes
- [x] includes documentation
- [ ] adds unit tests
- [x] adds or updates e2e tests

### CODE REVIEW

File: test/e2e/config/operator.yaml [hunk 1]
```diff
@@ -16,7 +16,8 @@ variables:
   RANCHER_FEATURES: "embedded-cluster-api=false"
   RANCHER_PATH: "rancher-stable/rancher"
   KUBERNETES_VERSION: "v1.26.3"
-  CAPI_INFRASTRUCTURE: "docker"
+  CAPI_INFRASTRUCTURE: "docker:v1.4.4"
```
Comment (richardcase) [inline test/e2e/config/operator.yaml:19]:
```suggestion
  CAPI_INFRASTRUCTURE: "docker:v1.4.6"
```

File: test/e2e/config/operator.yaml [hunk 2]
```diff
@@ -16,7 +16,8 @@ variables:
...
+  CAPI_CORE: "cluster-api:v1.4.4"
```
Comment (richardcase) [inline test/e2e/config/operator.yaml:20]:
```suggestion
  CAPI_CORE: "cluster-api:v1.4.6"
```

**Review by richardcase — COMMENTED** [review]
Looks good, just the one suggestion that we use 1.4.6 as thats the latest v1.4.x release

File: README.md [hunk 3]
```diff
@@ -66,6 +66,17 @@ What happens when you run `make dev-env`?
 4. `Rancher manager` is installed using helm.
 5. Run `tilt up` to start the development environment.
 
+# Testing
+
+We are using a combination of unit tests and e2e tests both using ginkgo and gomega frameworks.
+
+To run unit tests, execute:
```
Comment (furkatgofurov7) [inline README.md:73]:
```suggestion
To execute the tests, use the following command:
```

File: README.md [hunk 4]
```diff
@@ -66,6 +66,17 @@ What happens when you run `make dev-env`?
...
+```sh
+make test
+```
+
+Detailed documentation on e2e tests architecture and usage could be found [here](./test/e2e/README.md#e2e-tests).
```
Comment (furkatgofurov7) [inline README.md:78]:
```suggestion
Detailed documentation on e2e tests architecture and usage can be found [here](./test/e2e/README.md#e2e-tests).
```

**Review by furkatgofurov7 — CHANGES_REQUESTED** [review]

**Review by furkatgofurov7 — COMMENTED** [review]
Looks good, can we squash the commits?

**Comment by Danil-Grigorev** [comment]
@furkatgofurov7 We can always merge with squash. Prow setup will do this automatically.

File: test/e2e/config/operator.yaml [hunk 5]
```diff
@@ -16,7 +16,8 @@ variables:
   RANCHER_FEATURES: "embedded-cluster-api=false"
   RANCHER_PATH: "rancher-stable/rancher"
   KUBERNETES_VERSION: "v1.26.3"
-  CAPI_INFRASTRUCTURE: "docker"
+  CAPI_INFRASTRUCTURE: "docker:v1.4.6"
+  CAPI_CORE: "cluster-api:v1.4.6"
```
[the remark this answers is not in the document]
Comment (Danil-Grigorev) [reply test/e2e/config/operator.yaml:20]: We will have more then CAPD in e2e tests soon, it is better to have this included, as explicit is better then implicit, but thanks for pointing this out.

**Review by furkatgofurov7 — APPROVED** [review]

**Pull request merged**